File: scripts/auto-deploy-foundation.js
/**
 * AUTO DEPLOY Academy Foundation Tables
 * Uses direct PostgreSQL connection with service role
 */

import pg from 'pg';
import fs from 'node:fs';
import path from 'node:path';
import { fileURLToPath } from 'node:url';

const { Client } = pg;
const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Supabase PostgreSQL connection (Session mode - port 5432)
const client = new Client({
  host: 'aws-0-ap-southeast-1.pooler.supabase.com',
  port: 5432,
  database: 'postgres',
  user: 'postgres.diexsbzqwsbpilsymnfb',
  password: 'Acookingoil123',
  ssl: {
    rejectUnauthorized: false
  }
});

async function deployFoundation() {
  try {
    console.log('🚀 AUTO-DEPLOYING Academy Foundation Tables...\n');
    
    // Connect to database
    console.log('🔌 Connecting to Supabase PostgreSQL...');
    await client.connect();
    console.log('✅ Connected!\n');
    
    // Read SQL file
    const sqlPath = path.join(__dirname, '../supabase/migrations/20251114000001_academy_foundation_tables.sql');
    const sql = fs.readFileSync(sqlPath, 'utf8');
    
    console.log('📄 SQL file loaded');
    console.log('📏 Size:', (sql.length / 1024).toFixed(2), 'KB\n');
    
    console.log('⚡ Executing SQL...\n');
    
    // Execute SQL
    const result = await client.query(sql);
    
    console.log('✅ SQL executed successfully!\n');
    
    console.log('=' .repeat(60));
    console.log('🎉 DEPLOYMENT COMPLETE!');
    console.log('='.repeat(60));
    console.log('\n📋 Tables created:');
    console.log('  1. ✅ user_achievements');
    console.log('  2. ✅ user_xp');
    console.log('  3. ✅ study_groups');
    console.log('  4. ✅ study_group_members');
    console.log('  5. ✅ live_sessions');
    console.log('  6. ✅ live_session_attendees');
    console.log('  7. ✅ project_submissions');
    console.log('  8. ✅ student_revenue');
    console.log('\n🔧 Triggers: 4 created');
    console.log('📊 Views: 2 leaderboards created');
    console.log('🔒 RLS Policies: Enabled\n');
    console.log('🌟 Sample data:');
    console.log('  - 3 study groups');
    console.log('  - 4 upcoming live sessions\n');
    
  } catch (error) {
    console.error('\n❌ DEPLOYMENT FAILED:', error.message);
    console.error('\nFull error:');
    console.error(error);
    process.exit(1);
  } finally {
    await client.end();
    console.log('🔌 Connection closed.\n');
  }
}

// Run deployment
await deployFoundation();

File: scripts/auto_deploy_foundation.py
"""
AUTO DEPLOY Academy Foundation Tables to Supabase
Uses psycopg2 for direct PostgreSQL connection
"""

import psycopg2
import sys
from pathlib import Path

# Transaction pooler (US East 2)
CONNECTION_STRING = 'postgresql://postgres.diexsbzqwsbpilsymnfb:Acookingoil123@aws-1-us-east-2.pooler.supabase.com:6543/postgres'

def deploy_foundation():
    print('🚀 AUTO-DEPLOYING Academy Foundation Tables...\n')
    
    try:
        # Connect to database
        print('🔌 Connecting to Supabase PostgreSQL...')
        conn = psycopg2.connect(CONNECTION_STRING)
        cursor = conn.cursor()
        print('✅ Connected!\n')
        
        # Read SQL file
        sql_path = Path(__file__).parent.parent / 'supabase' / 'migrations' / '20251114000001_academy_foundation_tables.sql'
        with open(sql_path, 'r', encoding='utf-8') as f:
            sql = f.read()
        
        file_size = len(sql) / 1024
        print(f'📄 SQL file loaded')
        print(f'📏 Size: {file_size:.2f} KB\n')
        
        print('⚡ Executing SQL...\n')
        
        # Execute SQL
        cursor.execute(sql)
        conn.commit()
        
        print('✅ SQL executed successfully!\n')
        
        print('=' * 60)
        print('🎉 DEPLOYMENT COMPLETE!')
        print('=' * 60)
        print('\n📋 Tables created:')
        print('  1. ✅ user_achievements')
        print('  2. ✅ user_xp')
        print('  3. ✅ study_groups')
        print('  4. ✅ study_group_members')
        print('  5. ✅ live_sessions')
        print('  6. ✅ live_session_attendees')
        print('  7. ✅ project_submissions')
        print('  8. ✅ student_revenue')
        print('\n🔧 Triggers: 4 created')
        print('📊 Views: 2 leaderboards created')
        print('🔒 RLS Policies: Enabled\n')
        print('🌟 Sample data:')
        print('  - 3 study groups')
        print('  - 4 upcoming live sessions\n')
        
        cursor.close()
        conn.close()
        print('🔌 Connection closed.\n')
        
        return 0
        
    except psycopg2.Error as e:
        print(f'\n❌ DATABASE ERROR: {e}')
        print(f'Error Code: {e.pgcode}')
        print(f'Error Message: {e.pgerror}\n')
        return 1
    except FileNotFoundError as e:
        print(f'\n❌ FILE ERROR: {e}\n')
        return 1
    except Exception as e:
        print(f'\n❌ DEPLOYMENT FAILED: {e}\n')
        import traceback
        traceback.print_exc()
        return 1

if __name__ == '__main__':
    sys.exit(deploy_foundation())
